### PR TITLE
feat(storage.instance-backup): add public network to backup creation request payload

### PR DESCRIPTION
**Pull request by brofik**
ref: DTRSD-121379

| Question         | Answer
| ---------------- | ---
| Branch?          | `master` 
| Bug fix?         | yes
| New feature?     | no
| Breaking change? | no
| Tickets          | Fix #DTRSD-121379 
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] Breaking change is mentioned in relevant commits

### CODE REVIEW

File: packages/manager/modules/pci/src/projects/project/storages/instance-backups/add/add.component.js [hunk 1]
```diff
@@ -8,6 +8,7 @@ export default {
     excludeCategories: '<',
     backup: '<',
     goBack: '<',
+    publicNetworks: '<',
     publicNetwork: '<',
```
Comment (JacquesLarique) [inline packages/manager/modules/pci/src/projects/project/storages/instance-backups/add/add.component.js:12]: Is this still used, as you replaced it with publicNetworks in the routing ?

File: packages/manager/modules/pci/src/projects/project/storages/instance-backups/add/add.controller.js [hunk 2]
```diff
@@ -123,15 +124,25 @@ export default class PciInstancesAddController {
...
   }
 
+  isBarreMetalBackup() {
+    return !!this?.backup.type.includes(BAREMETAL_LABEL);
+  }
+
   onPrivateNetworkChange(modelValue) {
     const networkId = get(modelValue, 'id');
+    this.publicNetwork = this?.publicNetworks.find((network) => {
+      const networkName = this.isBarreMetalBackup()
```
Comment (ghyenne) [inline packages/manager/modules/pci/src/projects/project/storages/instance-backups/add/add.controller.js:134]:
```suggestion
      const networkName = this.isBareMetalBackup()
```

File: packages/manager/modules/pci/src/projects/project/storages/instance-backups/add/add.controller.js [hunk 3]
```diff
@@ -123,15 +124,25 @@ export default class PciInstancesAddController {
     this.messages = this.messageHandler.getMessages();
   }
 
+  isBarreMetalBackup() {
```
Comment (ghyenne) [inline packages/manager/modules/pci/src/projects/project/storages/instance-backups/add/add.controller.js:127]:
```suggestion
  isBareMetalBackup() {
```

File: packages/manager/modules/pci/src/projects/project/storages/instance-backups/add/add.controller.js [hunk 4]
```diff
@@ -123,15 +124,25 @@ export default class PciInstancesAddController {
...
+    return !!this?.backup.type.includes(BAREMETAL_LABEL);
+  }
+
   onPrivateNetworkChange(modelValue) {
     const networkId = get(modelValue, 'id');
+    this.publicNetwork = this?.publicNetworks.find((network) => {
```
Comment (JacquesLarique) [inline packages/manager/modules/pci/src/projects/project/storages/instance-backups/add/add.controller.js:133]: It doesn't seem necessary to use ? on **this**, maybe it's more relevant on **publicNetworks**

File: packages/manager/modules/pci/src/projects/project/storages/instance-backups/add/add.controller.js [hunk 5]
```diff
@@ -123,15 +124,25 @@ export default class PciInstancesAddController {
     this.messages = this.messageHandler.getMessages();
   }
 
+  isBarreMetalBackup() {
+    return !!this?.backup.type.includes(BAREMETAL_LABEL);
```
Comment (ghyenne) [inline packages/manager/modules/pci/src/projects/project/storages/instance-backups/add/add.controller.js:128]:
```suggestion
    return !!this.backup.type.includes(BAREMETAL_LABEL);
```

**Review by JacquesLarique — APPROVED** [review]

**Review by darsene — APPROVED** [review]

**Pull request merged**